### PR TITLE
Native CLI blog

### PR DESCRIPTION
**Pull request by ryanemerson**
My plan is to write a follow up post on how to use the native CLI with `kubectl`.

### CODE REVIEW

**Comment by ryanemerson** [comment]
https://github.com/infinispan/infinispan.github.io/pull/182

File: 2021-02-01-infinispan-cli-image.adoc [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+.. https://github.com/infinispan/infinispan-quarkus/releases/download/12.0.0.Final/infinispan-cli-12.0.0.Final-linux-amd64.zip[Linux]
+.. https://github.com/infinispan/infinispan-quarkus/releases/download/12.0.0.Final/infinispan-cli-12.0.0.Final-darwin-amd64.zip[Mac]
+.. https://github.com/infinispan/infinispan-quarkus/releases/download/12.0.0.Final/infinispan-cli-12.0.0.Final-windows-amd64.zip[Windows]
+
+. Start the CLI and connect to the server container:
+.. Linux & Mac: `./ispn-cli -c http://127.0.0.1:11222`
+.. Windows (Powershell): `& ".\ispn-cli.exe" -c http://<host>:11222`
+
+== Get it, Use it, Ask us!
+Please http://infinispan.org/download/[download],
```
Comment (pruivo) [inline 2021-02-01-infinispan-cli-image.adoc:46]:
```suggestion
Please https://infinispan.org/download/[download],
```

File: 2021-02-01-infinispan-cli-image.adoc [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+[start=2]
+. Download and extract the `.zip` for your desired platform:
+.. https://github.com/infinispan/infinispan-quarkus/releases/download/12.0.0.Final/infinispan-cli-12.0.0.Final-linux-amd64.zip[Linux]
+.. https://github.com/infinispan/infinispan-quarkus/releases/download/12.0.0.Final/infinispan-cli-12.0.0.Final-darwin-amd64.zip[Mac]
+.. https://github.com/infinispan/infinispan-quarkus/releases/download/12.0.0.Final/infinispan-cli-12.0.0.Final-windows-amd64.zip[Windows]
+
+. Start the CLI and connect to the server container:
+.. Linux & Mac: `./ispn-cli -c http://127.0.0.1:11222`
+.. Windows (Powershell): `& ".\ispn-cli.exe" -c http://<host>:11222`
```
Comment (pruivo) [inline 2021-02-01-infinispan-cli-image.adoc:43]: I would use http://127.0.0.1:11222 like in "linux & mac" command above.

File: 2021-02-01-infinispan-cli-image.adoc [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+[start=2]
+. Launch the CLI image and connect to the server image:
+
+----
+docker run --net=host -it --rm quay.io/remerson/cli -c http://127.0.0.1:11222
+----
+[start=3]
+. Enter the "user" and "pass" credentials, for the username and password respectively.
+
+TIP: The container's endpoint is the CLI binary, therefore it's possible to pass CLI args straight to the image,e.g. `infinispan/cli:12.0 --version`
```
Comment (pruivo) [inline 2021-02-01-infinispan-cli-image.adoc:30]:
```suggestion
TIP: The container's endpoint is the CLI binary, therefore it's possible to pass CLI args straight to the image,e.g. `quay.io/infinispan/cli:12.0 --version`
```

File: 2021-02-01-infinispan-cli-image.adoc [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+----
+docker run --net=host -it -p 11222:11222 -e USER="user" -e PASS="pass" quay.io/infinispan/server:12.0
+----
+
+==== Container
+[start=2]
+. Launch the CLI image and connect to the server image:
+
+----
+docker run --net=host -it --rm quay.io/remerson/cli -c http://127.0.0.1:11222
```
Comment (pruivo) [inline 2021-02-01-infinispan-cli-image.adoc:25]:
```suggestion
docker run --net=host -it --rm quay.io/infinispan/cli:12.0 -c http://127.0.0.1:11222
```

File: 2021-02-01-infinispan-cli-image.adoc [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+---
+
+Starting with Infinispan 12, we now provide a natively compiled version of the Infinispan CLI.
+
+This can be consumed as a native binary or via a container image.
+
+== Usage
+. Start a Infinispan server instance so that the CLI has an endpoint to connect to:
+----
+docker run --net=host -it -p 11222:11222 -e USER="user" -e PASS="pass" quay.io/infinispan/server:12.0
```
Comment (pruivo) [inline 2021-02-01-infinispan-cli-image.adoc:17]:
```suggestion
docker run --net=host -it --rm -p 11222:11222 -e USER="user" -e PASS="pass" quay.io/infinispan/server:12.0
```

File: 2021-02-01-infinispan-cli-image.adoc (hunk 3 above)
Comment (pruivo) [reply 2021-02-01-infinispan-cli-image.adoc:30]: maybe use the full command? `docker run --net=host --rm quay.io/infinispan/cli:12.0 --version`

File: 2021-02-01-infinispan-cli-image.adoc (hunk 5 above)
Comment (pruivo) [reply 2021-02-01-infinispan-cli-image.adoc:17]: `--net=host` isn't required for the server. `-p` forwards the port 11222 from the host machine to the container (at least on my machine)

File: 2021-02-01-infinispan-cli-image.adoc [hunk 6]
```diff
@@ -0,0 +1,49 @@
+---
+layout: blog
+title: Infinispan Native CLI
+permalink: /blog/:year/:month/:day/infinispan-12-final
```
Comment (oraNod) [inline 2021-02-01-infinispan-cli-image.adoc:4]:
```suggestion
permalink: /blog/:year/:month/:day/infinispan-cli-image
```

File: 2021-02-01-infinispan-cli-image.adoc (hunk 5 above)
Comment (ryanemerson) [reply 2021-02-01-infinispan-cli-image.adoc:17]: Good catch!

File: 2021-02-01-infinispan-cli-image.adoc [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+date: '2021-02-01T12:00:00.000-00:00'
+author: remerson
+tags: [ "docker", "cli", "containers", "quarkus", "native"]
+---
+
+Starting with Infinispan 12, we now provide a natively compiled version of the Infinispan CLI.
```
Comment (oraNod) [inline 2021-02-01-infinispan-cli-image.adoc:10]:
```suggestion
Starting with Infinispan 12, we provide a natively compiled version of the Infinispan CLI.
```

File: 2021-02-01-infinispan-cli-image.adoc [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+[start=2]
+. Launch the CLI image and connect to the server image:
+
+----
+docker run --net=host -it --rm quay.io/infinispan/cli:12.0 -c http://127.0.0.1:11222
+----
+[start=3]
+. Enter the "user" and "pass" credentials, for the username and password respectively.
+
+TIP: The container's endpoint is the CLI binary, therefore it's possible to pass CLI args straight to the image,e.g. `quay.io/infinispan/cli:12.0 --version`
```
Comment (oraNod) [inline 2021-02-01-infinispan-cli-image.adoc:30]:
```suggestion
TIP: The container's endpoint is the CLI binary, which lets you pass CLI arguments straight to the image, e.g. `quay.io/infinispan/cli:12.0 --version`
```

**Review by oraNod — COMMENTED** [review]
Couple minor comments if you don't mind a review. Great stuff, Ryan! This is awesome.

**Comment by ryanemerson** [comment]
@pruivo @oraNod Updated. Thanks for the review.

**Review by pruivo — APPROVED** [review]

File: _posts/2021-02-01-infinispan-cli-image.adoc [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+. Download and extract the `.zip` for your desired platform:
+.. https://github.com/infinispan/infinispan-quarkus/releases/download/12.0.0.Final/infinispan-cli-12.0.0.Final-linux-amd64.zip[Linux]
+.. https://github.com/infinispan/infinispan-quarkus/releases/download/12.0.0.Final/infinispan-cli-12.0.0.Final-darwin-amd64.zip[Mac]
+.. https://github.com/infinispan/infinispan-quarkus/releases/download/12.0.0.Final/infinispan-cli-12.0.0.Final-windows-amd64.zip[Windows]
+
+. Start the CLI and connect to the server container:
+.. Linux & Mac: `./ispn-cli -c http://127.0.0.1:11222`
+.. Windows (Powershell): `& ".\ispn-cli.exe" -c http://127.0.0.1:11222`
+
+== Get it, Use it, Ask us!
```
Comment (karesti) [inline _posts/2021-02-01-infinispan-cli-image.adoc:45]: remove this, not necessary

**Comment by ryanemerson** [comment]
Updated. Lets get this in and published.

**Comment by oraNod** [comment]
thanks @ryanemerson !

**Pull request merged**